compiler/rustc_middle/src/dep_graph/dep_node.rs

```diff
@@ -144,10 +144,7 @@ impl DepKind {
 
 macro_rules! define_dep_nodes {
     (
-    $(
-        [$($attrs:tt)*]
-        $variant:ident $(( $tuple_arg_ty:ty $(,)? ))*
-      ,)*
+        $( $( #[$attr:meta] )* $variant:ident, )+
```

Comment:
We lose the documentation for variants defined using the proc-macro, but get some for the nodes defined in this file. Shouldn't we have documentation for all or nothing?

Reply:
Oh, excellent catch. Yes, I forgot that previously `attrs` could include doc-comments, I agree we should keep those.

I almost wonder if we should remove `rustc_query_names` altogether and force everyone to match on the tokens for `rustc_query_append`. It's not *that* much extra work and it allows getting rid of another macro, and makes it easy to extend all the callsites to use the additional info in the future if they need it.

Reply:
Having just `rustc_query_append` seems great.  We'll just need a bit of creativity to give a signature to the 5 extra dep-nodes.  Actually, could we declare those 5 extra dep-nodes inside `rustc_middle::query` directly, and get rid of the `_append` part of the macro?

Reply:
Yes, I think I can do both those things :) and it should be possible to get rid of `rustc_cached_queries` at the same time, since we can match on the `cache` modifier in the macro rather than requiring it be done by the proc-macro.

Reply:
I'm going to avoid declaring the 5 dep-nodes in `rustc_middle::query` for now; ignoring the `snake_case` warning is harder than I expected, and it's not *that* hard to add the extra matchers in the other macros. I'll try and follow up on that later.

Reply:
Ok, got rid of `rustc_query_names` and `rustc_cached_queries`. I think it might be possible to remove `rustc_query_description` after https://github.com/rust-lang/rust/pull/101307 is merged, using `cache = $expr` and `desc = $fmt_str, $($arg),*`